go/teams/chain.go

```diff
@@ -2254,54 +2292,62 @@ func (t *teamSigchainPlayer) useInvites(stateToUpdate *TeamSigChainState, roleUp
-		if foundInvite {
-			isNewStyle, err := IsNewStyleInvite(invite)
-			if err != nil {
-				return err
-			}
-			if !isNewStyle {
-				return fmt.Errorf("`used_invites` for a non-new-style invite (id: %q)", inviteID)
+		inviteMD, foundInvite := stateToUpdate.FindActiveInviteMDByID(inviteID)
+		if !foundInvite {
+			if hasStubbedLinks {
+				// We didn't find it, possibly because server stubbed it out (we're not allowed to
+				// see it; didn't load with admin perms).
+				continue
+			} else {
+				// We couldn't find the invite, and we have no stubbed links, which
+				// means that inviteID is invalid.
+				return fmt.Errorf("could not find active invite ID in used_invites: %s", inviteID)
 			}
+		}
 
-			maxUses := invite.MaxUses
-			alreadyUsed := len(stateToUpdate.inner.UsedInvites[inviteID])
-			// Note that we append to stateToUpdate.inner.UsedInvites at the end of this for loop,
-			// so alreadyUsed updates correctly when processing multiple invite pairs.
-			if maxUses.IsUsedUp(alreadyUsed) {
-				return fmt.Errorf("invite %s is expired after %d uses", inviteID, alreadyUsed)
-			}
+		isNewStyle, err := IsNewStyleInvite(inviteMD.Invite)
+		if err != nil {
+			return err
+		}
+		if !isNewStyle {
+			return fmt.Errorf("`used_invites` for a non-new-style invite (id: %q)", inviteID)
+		}
 
-			// We explicitly don't check invite.Etime here; it's used as a hint for admins.
-			// but not checked in the sigchain player.
+		maxUses := inviteMD.Invite.MaxUses
+		alreadyUsed := len(inviteMD.UsedInvites)
+		// Note that we append to stateToUpdate.inner.UsedInvites at the end of this for loop,
+		// so alreadyUsed updates correctly when processing multiple invite pairs.
+		if maxUses.IsUsedUp(alreadyUsed) {
+			return fmt.Errorf("invite %s is expired after %d uses", inviteID, alreadyUsed)
+		}
 
-			// If we have the invite, also check if invite role matches role
-			// added.
-			var foundUV bool
-			for _, updatedUV := range roleUpdates[invite.Role] {
-				if uv.Eq(updatedUV) {
-					foundUV = true
-					break
-				}
-			}
-			if !foundUV {
-				return fmt.Errorf("used_invite for UV %s that was not added as role %s", pair.UV, invite.Role.HumanString())
+		// We explicitly don't check invite.Etime here; it's used as a hint for admins.
+		// but not checked in the sigchain player.
+
+		// If we have the invite, also check if invite role matches role
+		// added.
+		var foundUV bool
+		for _, updatedUV := range roleUpdates[inviteMD.Invite.Role] {
+			if uv.Eq(updatedUV) {
+				foundUV = true
+				break
 			}
-		} else if !hasStubbedLinks {
-			// We couldn't find the invite, and we have no stubbed links, which
-			// means that inviteID is invalid.
-			return fmt.Errorf("could not find active invite ID in used_invites: %s", inviteID)
+		}
+		if !foundUV {
+			return fmt.Errorf("used_invite for UV %s that was not added as role %s", pair.UV,
+				inviteMD.Invite.Role.HumanString())
 		}
 
 		logPoint := len(stateToUpdate.inner.UserLog[uv]) - 1
```

Comment:
made me thing: can you add a comment that because of how `UserLog` is referenced here, `useInvites` is called after `updateMembership`?